wherescape/connectors/gitlab/python_gitlab_high_water_mark.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+import logging
+
+from ... import WhereScape
+
+
+def gitlab_next_high_water_mark():
+    """
+    At the beginning of the load data job, this function will set the next
+    high water mark. As the high water mark date value is shared over all
+    Gitlab objects, this needs to be determined before the first object is
+    being loaded. This date value will be stored seperately until the job
+    has finished.
+    """
+    wherescape_instance = WhereScape()
+    next_high_water_mark = datetime.today().isoformat(timespec="seconds")
+    logging.info(f"Next high water mark will be {next_high_water_mark}")
+    wherescape_instance.main_message = (
+        f"Next high water mark will be {next_high_water_mark}"
+    )
+    wherescape_instance.write_parameter(
+        "gitlab_high_water_mark_next", next_high_water_mark
+    )
+
+
+def gitlab_update_high_water_mark():
```

Comment:
It seems to do something to the lines of [spindle_postgres_pscript_post](https://gitlab.wearespindle.com/data/wherescape-warehouse/-/blob/c4b047e8acad7ea65e635a5a3c2143365046e051/pebble/spindle_postgres_pscript_post.peb) but then differently?